client/dive-common/use/useSave.ts

```diff
@@ -65,24 +65,30 @@ export default function useSave(datasetId: Ref<Readonly<string>>) {
   function markChangesPending(
     {
       action,
-      data,
+      track,
+      attribute,
     }: {
       action: 'upsert' | 'delete' | 'meta';
-      data?: Track | Attribute;
+      track?: Track;
+      attribute?: Attribute;
     } = { action: 'meta' },
   ) {
     if (action === 'meta') {
       pendingChangeMap.meta += 1;
-    } else if (isTrack(data)) {
+    } else if (track !== undefined) {
       _updatePendingChangeMap(
-        data.trackId, data, action, pendingChangeMap.upsert, pendingChangeMap.delete,
+        track.trackId, track, action, pendingChangeMap.upsert, pendingChangeMap.delete,
       );
-    } else if (isAttribute(data)) {
+    } else if (attribute !== undefined) {
       _updatePendingChangeMap(
-        data.key, data, action, pendingChangeMap.attributeUpsert, pendingChangeMap.attributeDelete,
+        attribute.key,
+        attribute,
+        action,
+        pendingChangeMap.attributeUpsert,
+        pendingChangeMap.attributeDelete,
       );
     } else {
-      throw new Error(`Arguments inconsistent with pending change type: ${action} cannot be performed on ${data}`);
+      throw new Error(`Arguments inconsistent with pending change type: ${action} cannot be performed on ${attribute}`);
```

Comment:
`attribute` will always be undefined here, maybe a slight change to this message.